services/common/src/components/projectSummary/DocumentUpload.tsx

```diff
@@ -63,15 +43,12 @@ export const DocumentUpload: FC<DocumentUploadProps> = (props) => {
   };
 
   useEffect(() => {
-    props.change(FORM.ADD_EDIT_PROJECT_SUMMARY, "spatial_documents", []);
-    props.change(FORM.ADD_EDIT_PROJECT_SUMMARY, "support_documents", []);
-  }, []);
-
-  useEffect(() => {
-    props.change(FORM.ADD_EDIT_PROJECT_SUMMARY, "documents", [
-      ...spatial_documents,
-      ...support_documents,
-    ]);
+    dispatch(
+      change(FORM.ADD_EDIT_PROJECT_SUMMARY, "documents", [
+        ...spatial_documents,
+        ...support_documents,
+      ])
+    );
```

Comment:
Making sure that documents persist between user navigation events and are added to the table when uploaded. spatial & support variables are populated right in formatProjectSummary